### PR TITLE
[SPARK-41754][UI] Add simple developer guides for UI Protobuf serializer

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

* Add simple developer guides for UI Protobuf serializer.
* Rename `UNSPECIFIED` in the enum JobExecutionStatus as `JOB_EXECUTION_STATUS_UNSPECIFIED` 

### Why are the changes needed?

1. Provide a simple guide for common issues found during PR review
2. Avoid naming conflicts in enum definitions. 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing UT

### CODE REVIEW

**Comment by gengliangwang** [comment]
This is for the issue in https://github.com/apache/spark/pull/39192#discussion_r1058002256
cc @LuciferYang @panbingkun

**Review by cloud-fan — APPROVED** [review]

File: core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto [hunk 1]
```diff
@@ -18,8 +18,15 @@
 syntax = "proto3";
 package org.apache.spark.status.protobuf;
 
+/**
+ * Developer guides:
+ *   - Coding style: https://developers.google.com/protocol-buffers/docs/style
+ *   - Use int64 for job/stage IDs, in case of future extension in Spark core.
+ *   - Use `weakIntern` on string values in create new objects during deserialization.
```
Comment (LuciferYang) [inline core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto:25]: `- Use `weakIntern` on string values in create new objects during deserialization.`

This is very important. I think we need a refactor work to fix the current codes

Comment (LuciferYang) [reply core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto:25]: cc @panbingkun @techaddict

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

File: core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto (hunk 1 above)
Comment (panbingkun) [reply core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto:25]: https://issues.apache.org/jira/browse/SPARK-41759 @LuciferYang @gengliangwang

**Review by panbingkun — APPROVED** [review]

**Comment by LuciferYang** [comment]
need re-trigger failed GA task

File: core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto [hunk 2]
```diff
@@ -18,8 +18,15 @@
...
+ */
+
 enum JobExecutionStatus {
-  UNSPECIFIED = 0;
+  JOB_EXECUTION_STATUS_UNSPECIFIED = 0;
```
Comment (panbingkun) [inline core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto:29]: Is it more reasonable?

message Job {
   enum JobExecutionStatus {
      ...
   }
}

**Review by techaddict — APPROVED** [review]

File: core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto (hunk 2 above)
Comment (panbingkun) [reply core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto:29]: OK, Let us follow code style(https://developers.google.com/protocol-buffers/docs/style#enums):
image

**Comment by gengliangwang** [comment]
Merging to master

**Pull request closed without merging**